src/tree/cache.rs

```diff
@@ -171,10 +175,15 @@ impl Cache {
         }
     }
 
-    /// Clear all cache entries
-    pub fn clear(&mut self) {
+    /// Clear all cache entries and reports if they were already cleared
+    pub fn clear(&mut self) -> bool {
```

Comment:
Perhaps we should consider an `enum` (`Cleared` | `AlreadyEmpty`). As I think it is not obvious what the boolean represents here (it could easily be the other way around!)